algorithms/indexing/indexer.py

```diff
@@ -571,6 +571,8 @@ def index(self):
                 # no more lattices found
                 break
 
+            refined_experiments = None
+            refined_reflections = None
```

Comment:
This change is actually in the wrong loop level, and breaks the very test that was designed to exercise the relevant try/except loop:
```
$ pytest algorithms/indexing/test_index.py::test_refinement_failure_on_max_lattices_a15 --regression
...
Traceback (most recent call last):
  File "/Users/rjgildea/software/cctbx_py3/build/../../cctbx/modules/dials/command_line/index.py", line 261, in <module>
    run()
  File "/Users/rjgildea/software/cctbx_py3/conda_base/python.app/Contents/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/Users/rjgildea/software/cctbx_py3/build/../../cctbx/modules/dials/command_line/index.py", line 244, in run
    indexed_experiments, indexed_reflections = index(
  File "/Users/rjgildea/software/cctbx_py3/build/../../cctbx/modules/dials/command_line/index.py", line 162, in index
    indexed_experiments, indexed_reflections = _index_experiments(
  File "/Users/rjgildea/software/cctbx_py3/build/../../cctbx/modules/dials/command_line/index.py", line 114, in _index_experiments
    idxr.index()
  File "/Users/rjgildea/software/cctbx/modules/dials/algorithms/indexing/indexer.py", line 659, in index
    sel = refined_reflections["id"] == last
TypeError: Please report this error to dials-support@lists.sourceforge.net: 'NoneType' object is not subscriptable
```

https://github.com/dials/dials/blob/fcfd7b452e93a82e718aba9663e5712e0244a821/algorithms/indexing/test_index.py#L567-L604